lib/knock/client.rb

```diff
@@ -6,16 +6,22 @@ module Client
     include Kernel
 
     def client
-      return @client if defined?(@client)
-
-      @client = Net::HTTP.new(Knock::API_HOSTNAME, 443)
-      @client.use_ssl = true
-
-      @client
+      Net::HTTP.new(Knock.config.api_hostname, 443).tap do |http_client|
+        http_client.use_ssl = true
+        http_client.open_timeout = Knock.config.timeout
+        http_client.read_timeout = Knock.config.timeout
+        http_client.write_timeout = Knock.config.timeout if RUBY_VERSION >= '2.6.0'
+      end
```

Comment:
This will create a new client for every request. While it will work it is not ideal for performance. 

As an alternate you could use a connection pool via [connection pool gem](https://github.com/mperham/connection_pool) and that way you can reuse the connections. Something like the following: 

```
@pool = ConnectionPool.new(size: 5) { client }


pool.with |client| do
  client.request(request)
end
```

If you do decide to create a pool then it would be good to also have a configuration to use [sessions](https://docs.ruby-lang.org/en/3.2/Net/HTTP.html#class-Net::HTTP-label-HTTP+Sessions) and then you can include a call to `start` when you create a client. 

```
http_client.start if Knock.config.use_sessions
```